app/src/main/java/com/practicum/myplaylistmaker/ui/settings/view_model/SettingsViewModel.kt

```diff
@@ -0,0 +1,75 @@
+package com.practicum.myplaylistmaker.ui.settings.view_model
+
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.practicum.myplaylistmaker.App.App
+import com.practicum.myplaylistmaker.domain.settings.SettingsInteractor
+import com.practicum.myplaylistmaker.domain.sharing.SharingInteractor
+import com.practicum.myplaylistmaker.util.Creator
+
+class SettingsViewModel(
+    private var sharingInteractor: SharingInteractor,
+    private var settingsInteractor: SettingsInteractor
+) : ViewModel() {
+    init {
+        sharingInteractor = Creator.provideSharingIneractor()
+        settingsInteractor = Creator.provideSettingInteractor()
+    }
+
+    private var onBackLiveData = MutableLiveData(false)
```

Comment:
```suggestion
    private val onBackLiveData = MutableLiveData(false)
```
Переменной нигде не присваивается новое значение, значит можно как `val` объявлять. Все переменные нужно стараться объявлять как val